src/services/AsyncStorage.ts

```diff
@@ -140,6 +141,40 @@ export const getDevMode = async (): Promise<boolean | null> => {
   return isDevMode ? JSON.parse(isDevMode) : null
 }
 
+export const getUserVocabulary = async (): Promise<UserVocabularyDocument[]> => {
+  const userVocabulary = await AsyncStorage.getItem(USER_VOCABULARY)
+  return userVocabulary ? JSON.parse(userVocabulary) : []
+}
+
+const setUserVocabulary = async (userDocument: UserVocabularyDocument[]): Promise<void> => {
+  await AsyncStorage.setItem(USER_VOCABULARY, JSON.stringify(userDocument))
+}
+
+const addUserDocument = async (userDocument: UserVocabularyDocument): Promise<void> => {
+  const userVocabulary = await getUserVocabulary()
+  await setUserVocabulary([...userVocabulary, userDocument])
+}
+
+const editUserDocument = async (
+  oldUserDocument: UserVocabularyDocument,
+  newUserDocument: UserVocabularyDocument
+): Promise<boolean> => {
+  const userVocabulary = await getUserVocabulary()
+  const index = userVocabulary.findIndex(item => JSON.stringify(item) === JSON.stringify(oldUserDocument))
+  if (index === -1) {
+    return false
+  }
+  userVocabulary[index] = newUserDocument
+  await setUserVocabulary(userVocabulary)
+  return true
+}
+
+const deleteUserDocument = async (userDocument: UserVocabularyDocument): Promise<void> => {
+  let userVocabulary = await getUserVocabulary()
+  userVocabulary = userVocabulary.filter(item => JSON.stringify(item) !== JSON.stringify(userDocument))
+  await setUserVocabulary(userVocabulary)
```

Comment:
No need to use `let` here, either inline or create a separate var.
```suggestion
  const userVocabulary = await getUserVocabulary()
  const newUserVocabulary = userVocabulary.filter(item => JSON.stringify(item) !== JSON.stringify(userDocument))
  await setUserVocabulary(newUserVocabulary)
```